src/components/JoinPage/JoinAgree/index.tsx

```diff
@@ -0,0 +1,164 @@
+    },
+    {
+      id: 4,
+      state: false,
+      level: 3,
+      checkBox: <IcEmptyCheckBox />,
+      text: "마케팅 활용/광고성 정부 수신동의 (선택)",
+      button: <IcNextBtn />,
+    },
+    { id: 5, state: false, level: 2, checkBox: <IcEmptyCheckBox />, text: "만 14세 이상입니다 (필수)" },
+  ]);
+
+  useEffect(() => {
+    console.log(lists);
+    console.log(lists[isPicked]);
+    if (lists[isPicked].state === false) {
+      lists[isPicked].state = true;
+      setCheckActive(true);
+    } else if (lists[isPicked].state === true) {
+      lists[isPicked].state = false;
+      setCheckActive(false);
+    }
+  }, [isPicked, lists]);
+
+  function ChangeCheckBox(index: number) {
+    const currentList = [...lists];
+    setIsPicked(index);
+
+    if (currentList[index].state === false) {
+      currentList[index].checkBox = <IcFullCheckBox />;
+      setCheckActive(false);
+    } else if (currentList[index].state === true) {
+      currentList[index].checkBox = <IcEmptyCheckBox />;
+      setCheckActive(true);
+    }
+    alertClassName;
+    setLists(currentList);
+
+    //handleAllCheck(currentList[0].state);
+    //CheckAllPicked();
+
+    // if (currentList[0].state) {
+    //   CheckAllPicked();
+    //
+    //   if (checkItems.length != 4) {
+    //     console.log("1");
+    //     handleAllCheck(false);
+    //   } else {
+    //     console.log("2");
+    //     handleAllCheck(true);
+    //     console.log("배열 길이", checkItems.length);
+    //   }
+    // }
+  }
+
+  /*
+  function CheckAllPicked() {
+    const currentList = [...lists];
+
+    if (allPicked === false) {
+      //currentList[0].checkBox = <IcFullCheckBox />;
+      currentList[1].checkBox = <IcFullCheckBox />;
+      currentList[2].checkBox = <IcFullCheckBox />;
+      currentList[3].checkBox = <IcFullCheckBox />;
+      currentList[4].checkBox = <IcFullCheckBox />;
+      //currentList[0].state = true;
+      currentList[1].state = true;
+      currentList[2].state = true;
+      currentList[3].state = true;
+      currentList[4].state = true;
+    } else {
+      //currentList[0].checkBox = <IcEmptyCheckBox />;
+      currentList[1].checkBox = <IcEmptyCheckBox />;
+      currentList[2].checkBox = <IcEmptyCheckBox />;
+      currentList[3].checkBox = <IcEmptyCheckBox />;
+      currentList[4].checkBox = <IcEmptyCheckBox />;
+      //currentList[0].state = false;
+      currentList[1].state = false;
+      currentList[2].state = false;
+      currentList[3].state = false;
+      currentList[4].state = false;
+    }
+  }
+
+  const handleAllCheck = (checked: boolean) => {
+    if (checked) {
+      const idArray: Array<number> = [];
+      lists.forEach((el) => idArray.push(el.id));
+      setCheckItems(idArray);
+      setAllPicked(true);
+      console.log("배열 길이", checkItems.length);
+    } else {
+      setCheckItems([]);
+      setAllPicked(false);
+    }
+  };
+
+  */
+
+  function alertClassName() {
+    return openAlert === false ? "login-alert" : "login-alert-view";
+  }

```

Comment:
👀반영 제안👀

```suggestion
  const alertClassName = openAlert === false ? "login-alert" : "login-alert-view";
```

이렇게 선언하면 되지 않을까 합니다!